src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (
+      victim instanceof Animal &&
+      victim.hasOwnProperty('hidden') &&
+      !victim.hidden
+    ) {
+      victim.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
The use of `filter` here is a good approach to handle dead animals. However, ensure that the condition `animal.health > 0` correctly filters out only those animals whose health is greater than zero. This line should effectively remove any animal with health less than or equal to zero from the `Animal.alive` array.